### ISSUE TITLE
1.7.0 display issues on mobile

### ISSUE THREAD

**Issue by WizWareTech** [cuzy-app/humhub-modules-clean-theme#11]
1.6.0 worked great.

1.7.0 has some nice new features, but here is feedback as to why we reverted back to 1.6.0 for now:

The split menu is bad for us for these reasons:
1) The names disappear for the menu items, and the users don't know what does what.
2) The site name ends up wrapping to two lines at the top, causing it to float above content below the menu.

If we move to 1.7.0 we must have a way to disable the split menu and go back to the 1.6.0 functionality for the menu. Can you provide an option in the config file to disable the split menu feature?

> Enh: On small screens and on scrolling down, the top and bottom menus are hidden (this feature can be disabled in the configuration file)

How do we disable this feature in the configuration file?

If these items can be addressed, we are excited to move to 1.7.0 and I think others will also appreciate this too.

**Comment by marc-farre** [comment]
@WizWareTech Thanks for your feedback.
Can you provide screenshots of the issues? Thanks.

**Comment by ArchBlood** [comment]
@marc-farre can you update the title for v1.7.1 as I've found another issue when it comes to the tasks module as the menu completely disappears for all mobile devices.

**Comment by marc-farre** [comment]
Thanks @ArchBlood However, I cannot reproduce:
![image](https://github.com/cuzy-app/humhub-modules-clean-theme/assets/23310825/d25b98bd-b832-4ca2-9a3e-5b5bbc8d1668)

Have you flushed the cache after installing the new version?

**Comment by WizWareTech** [comment]
@marc-farre yes I can do this. But it is important that I am able to disable the split menu and return the menu functionality to how 1.6x works. (I need the text showing along with icons, not just icons alone). Is there a config option I can set to:
1) Include text with icon in menus
2) Disable split menu

Those two options will be very important for us. Our users have enough trouble knowing what something does even with the words there. With only an icon, they are clueless.

**Comment by marc-farre** [comment]
@WizWareTech I like the idea of displaying the text below the icons also on the bottom bar. I'll add them as an option in a few days.

But why disabling the split menu?
The bottom menu is common with apps and avoids the dropdown menu: we now have direct access to menu items, which makes navigation much better on mobile.

**Comment by ArchBlood** [comment]
> Thanks @ArchBlood However, I cannot reproduce:
> ![image](https://github.com/cuzy-app/humhub-modules-clean-theme/assets/23310825/d25b98bd-b832-4ca2-9a3e-5b5bbc8d1668)
> 
> Have you flushed the cache after installing the new version?

Using the Custom Pages module to create pages that append to the top menu, navigate to the tasks overview and the issue will be shown.

**Comment by WizWareTech** [comment]
The reason for not wanting split menu is because when describing to users where to find menu items, now the menu items are sometimes on the top, sometimes on the bottom, etc. Many users are not sophisticated, and get easily confused when things move around (especially when moving from dekstop to mobile in portrait mode). 

I think more advanced users will have no problem with split menu, but for one of our customers, the users are over 50 yrs old and struggle to learn things in the first place. Is there a way to have a switchable config parameter to enable/disable split menu? That way both bases are covered.

**Comment by marc-farre** [comment]
@ArchBlood Thanks. I'll have a try soon.

@WizWareTech  Thanks for the explanation. I understand, but I don't have this need, and I think the advantages outweigh the disadvantages.
And people are getting more and more used with the dropdown menu as it's very common now with mobile apps.
And the goal of this theme is to fit well with HumHub mobile app.
Moreover, I don't want to maintain 2 versions of the navigation menu. Too much work for a free module, sorry.

**Comment by ArchBlood** [comment]
> @ArchBlood Thanks. I'll have a try soon.
> 
> @WizWareTech Thanks for the explanation. I understand, but I don't have this need, and I think the advantages outweigh the disadvantages. And people are getting more and more used with the dropdown menu as it's very common now with mobile apps. And the goal of this theme is to fit well with HumHub mobile app. Moreover, I don't want to maintain 2 versions of the navigation menu. Too much work for a free module, sorry.

Maybe append to `.layout-content-container`? I do this for the scroll up module.

**Comment by WizWareTech** [comment]
@marc-farre If the text names can be added to the menu items (even a switchable parameter) then even with split, it would be MUCH better. I hope you can make that happen. I am happy to test and give you feedback/screenshots, just let me know.

**Comment by ArchBlood** [comment]
> @marc-farre If the text names can be added to the menu items (even a switchable parameter) then even with split, it would be MUCH better. I hope you can make that happen. I am happy to test and give you feedback/screenshots, just let me know.

Appending text based names in this situation is possible but for a loss of mobile friendliness on newer devices with smaller screens, and adding tooltips is almost impossible currently for mobile, it would be much easier to breakdown each menu item for your users and explain what does what than doing this, and in all honesty, if text were added it would be very small in the current view which isn't mobile friendly.

I can agree with @marc-farre that maintaining two separate navigation menus would require too much time and the positives outweigh the negatives here, if you could get more people onboard with this suggestion then that would be a different story.

**Comment by marc-farre** [comment]
> Currently this is what I see in my tests;

@ArchBlood Sorry, I cannot reproduce.

1/ I've created a Markdown custom page.

2/ I went to it via the bottom menu:
![image](https://github.com/cuzy-app/humhub-modules-clean-theme/assets/23310825/8ed7384b-bdd0-49c2-9057-4320647c72c6)

3/ I went to the Tasks page:
![image](https://github.com/cuzy-app/humhub-modules-clean-theme/assets/23310825/75b9fda7-4470-4223-b7a9-b3984abab301)

What I can see, is that the "Tasks" page is not responsive: a horizontal scrollbar is added because the page width is higher than the screen one. Maybe it should be fixed in the Tasks module?

**Comment by ArchBlood** [comment]
> > Currently this is what I see in my tests;
> 
> @ArchBlood Sorry, I cannot reproduce.
> 
> 1/ I've created a Markdown custom page.
> 
> 2/ I went to it via the bottom menu:
> ![image](https://github.com/cuzy-app/humhub-modules-clean-theme/assets/23310825/0bf377ef-2fb4-4e7e-ab6c-20e80b9c00e4)
> 
> 3/ I went to the Tasks page:
> ![image](https://github.com/cuzy-app/humhub-modules-clean-theme/assets/23310825/75b9fda7-4470-4223-b7a9-b3984abab301)
> 
> What I can see, is that the "Tasks" page is not responsive: a horizontal scrollbar is added because the page width is higher than the screen one. Maybe it should be fixed in the Tasks module?

You are absolutely right, it seems like the tasks module is the issue when it comes to this.

**Comment by marc-farre** [comment]
@WizWareTech done in commit https://github.com/cuzy-app/humhub-modules-clean-theme/commit/424a9f60fed05e3599cb906ca691e771c88ed876 and https://github.com/cuzy-app/humhub-modules-clean-theme/commit/7722160eb3f370a6ff889df6a6e2032c3e006070

I've added the module configuration `hideTextInBottomMenuItems`
See https://docs.humhub.org/docs/admin/advanced-configuration#module-configurations to configure it.
You must set this config to false.

Let me know what you think about it.
It looks like that:
![image](https://github.com/cuzy-app/humhub-modules-clean-theme/assets/23310825/955acf06-e738-4d0a-8fd2-7e2b0d48a87c)

**Comment by WizWareTech** [comment]
@marc-farre I'm not versed in using git. Can you tell me what commands I need in order to pull those two commits in to my clean-theme folder?

**Comment by WizWareTech** [comment]
@marc-farre 

I just pulled the entire thing with git clone for now.

Three issues:
1) The name of application (website name) gets moved to two lines:
![image](https://github.com/cuzy-app/humhub-modules-clean-theme/assets/13114/fb3d429e-d1bc-4579-9e41-50feb5ff1e26)

Where can I adjust/reduce the font size to fit this screen?

2) The second line stays there during scrolling:
![image](https://github.com/cuzy-app/humhub-modules-clean-theme/assets/13114/abf07b5e-587b-4094-81ed-543cdd51a27c)

3) 'DASHBOARD' gets cut into two at the bottom:
![image](https://github.com/cuzy-app/humhub-modules-clean-theme/assets/13114/48c24b4c-927e-4370-97df-87ef2aebb7e9)

Can I reduce this font size as well? (or only for longer items perhaps?)

**Comment by marc-farre** [comment]
@WizWareTech thanks for your feedback!

> 1. The name of application (website name) wraps to two lines

Thanks. Fixed in commit https://github.com/cuzy-app/humhub-modules-clean-theme/commit/a1d007ee6eab304d9a3b69cd05366702637adb05

> 3. 'DASHBOARD' gets cut into two at the bottom

Yes. What do you suggest?
- Smaller font size is too small.
- If I add more width to the menu item, then I need to add width also to the others to have the same width, which will end with less visible items (more items will be in the right drop up menu).

> 4. The vertical spacing between items in the My Space menu is far too wide.

Yes. Fixed in commit https://github.com/cuzy-app/humhub-modules-clean-theme/commit/67c10e3f563fb05e13bfbb5f7ad2a83fef8b0ca9

> I just pulled the entire thing with git clone for now.

Yes, that's fine.
Now you can update with the latest fixes with a `git pull`

If you change something in the `assets` folder, it will be erased next time you flush the cache.
So:
- if it's about issues or enhance ideas that affect all users of the Clean Theme, create a new issue on https://github.com/cuzy-app/humhub-modules-clean-theme/issues and I will fix it for everybody and maintain it in future versions
- if you have specific needs for your HumHub instance (special colors, special font, etc), then create a child theme: https://github.com/cuzy-app/humhub-modules-clean-theme?tab=readme-ov-file#child-themes

**Comment by WizWareTech** [comment]
Perhaps I'm doing this wrong. I ran 'git clone https://github.com/cuzy-app/humhub-modules-clean-theme.git' to get the most recent version. Then I moved it into modules as 'clean-theme'.

Then I started getting 404 errors on resources, for example:

404 Not Found
https://XXXXX.XX/assets/6869c2d/less/home/marc/www/hhdev/static/less/gridview.less

Is there something else I need to do once performing a git clone or git pull to update these assets? And why is it looking in a 'marc' folder? The 'home' folder doesn't exist under that asset folder. For instance, I can see /assets/6869c2d/less, but there is no 'home' folder in there.

**Comment by marc-farre** [comment]
@WizWareTech You shouldn't put anything into the assets folder. Files are auto-generated from cache.

The Clean Theme module must be in `protected/modules/clean-theme`.

So:
```
cd protected/modules/
rm -rf clean-theme
git clone https://github.com/cuzy-app/humhub-modules-clean-theme.git clean-theme
```

Then, flush the cache: Administration -> Settings -> Advanced -> "Save and Flush cache".
This will rebuild the assets folder from the new module version.

**Comment by WizWareTech** [comment]
I wasn't putting anything in assets, but I noticed files were missing from there, which makes sense why, now.

The flushing is what was needed to rebuild assets and now it is working. Thank you, and sorry for my lack of knowledge in the right steps to manually update the module. I wish this was explained in HumHub docs. I'll continue testing the new commits and give you feedback.

**Comment by ArchBlood** [comment]
> I wasn't putting anything in assets, but I noticed files were missing from there, which makes sense why, now.
> 
> The flushing is what was needed to rebuild assets and now it is working. Thank you, and sorry for my lack of knowledge in the right steps to manually update the module. I wish this was explained in HumHub docs. I'll continue testing the new commits and give you feedback.

It is explained a little bit in the HumHub Docs;
https://docs.humhub.org/docs/develop/build/

**Comment by WizWareTech** [comment]
@marc-farre 

The latest changes look much better with the spacing!

1. For the 'DASHBOARD' wrapping to new line, I have an idea. Can the 'DASHBOARD' be translated to 'DASH-BOARD' and if so, will that cause it to wrap after the 'DASH-'? If so, can you tell me how to change the translation for it? I searched by am unable to locate the corresponding module to make the translation, following guide in https://docs.humhub.org/docs/admin/translations/.

2. When going into a space, the space name disappears. This is confusing to users. Is there a way to put the space name in there even if some of it gets cut off due to size restriction?

![image](https://github.com/cuzy-app/humhub-modules-clean-theme/assets/13114/ef345b67-2a46-4faa-b79f-993b3659398b)

And thank you for your great work, and fast response, it is much appreciated!

**Comment by ArchBlood** [comment]
> @marc-farre
> 
> The latest changes look much better with the spacing!
> 
> 1. For the 'DASHBOARD' wrapping to new line, I have an idea. Can the 'DASHBOARD' be translated to 'DASH-BOARD' and if so, will that cause it to wrap after the 'DASH-'? If so, can you tell me how to change the translation for it? I searched by am unable to locate the corresponding module to make the translation, following guide in https://docs.humhub.org/docs/admin/translations/.
> 2. When going into a space, the space name disappears. This is confusing to users. Is there a way to put the space name in there even if some of it gets cut off due to size restriction?
> 
> ![image](https://private-user-images.githubusercontent.com/13114/314373106-ef345b67-2a46-4faa-b79f-993b3659398b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA5MTI2NjQsIm5iZiI6MTcxMDkxMjM2NCwicGF0aCI6Ii8xMzExNC8zMTQzNzMxMDYtZWYzNDViNjctMmE0Ni00ZmFhLWI3OWYtOTkzYjM2NTkzOThiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzIwVDA1MjYwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZjNmFjMjkyZTg2MGJmNzcyZTk2OGE4NzEzMDFiYWZiMzgxY2IxY2U5OWM1YjJjODAwMDJjNjVjMjI2ZjQ5MjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.WYe7iEOG1xTko-2sROWGj-c1WYSgCWaPJe7UzeWr-Iw)
> 
> And thank you for your great work, and fast response, it is much appreciated!

Answered in the community;
https://community.humhub.com/comment/perma?id=48323

**Comment by marc-farre** [comment]
@WizWareTech Fixed in commit https://github.com/cuzy-app/humhub-modules-clean-theme/commit/39da6f8632ba94be6020774df6e0edd5792de2ef

I've also added more space so that "DASHBOARD" don't display in 2 lines (but we might have the issue with other menu entries and other languages).

![image](https://github.com/cuzy-app/humhub-modules-clean-theme/assets/23310825/0150fa3c-9dd5-4396-8a96-54c5abf38753)

To update:
```sh
cd protected/modules/clean-theme
git pull
```
Administration -> Settings -> Advanced -> "Save and Flush cache".

Tell me if everything's fine now. Then I'll release a new version of the module. Thanks!

**Comment by WizWareTech** [comment]
I see this now:
![image](https://github.com/cuzy-app/humhub-modules-clean-theme/assets/13114/4e068587-72fa-4369-b626-0f41856004db)

And when selecting a space:
![image](https://github.com/cuzy-app/humhub-modules-clean-theme/assets/13114/fa95f006-24be-40c5-a384-d00c87f2f56e)

Is there a way to include the name of the space below the selected space icon? (I understand that there will be length issues there, so I assume you'll have to truncate using maybe text-overflow? I'm not a css expert)

**Comment by marc-farre** [comment]
@WizWareTech can you:
- either wait for the 1.7.2 to be available on the marketplace and update to it
- or download it from https://marketplace.humhub.com/module/clean-theme/downloads and flush the cache
And then refresh your browser page?

If you still have the issue after doing this, let me know the browser name and version you are using, and the OS (Android, etc.).

The Space name is not available because the theme only change the style and on the native HumHub the name is not displayed:
![image](https://github.com/cuzy-app/humhub-modules-clean-theme/assets/23310825/cbebf0b1-cd05-4a29-8da4-452bfea67a47)

**Comment by marc-farre** [comment]
@WizWareTech Version 1.7.2 is now available. Just update using the integrated marketplace.

**Comment by WizWareTech** [comment]
Another issue:

When searching under the my spaces button, if there are a full screen of results:
1) They fall off the top edge and you can't see anything above
2) the input field is also off the top of the screen
3) the on screen keyboard disappears
4) when attempting to scroll the results up/down, the entire page moves instead of the pop-up result menu

Proposed solution:
limit results that show to allow the input field to remain on the screen at the top, and keep the on screen keyboard on screen so that characters can continue being typed in to narrow the search further. also, fix the scrolling so the results can be scrolled within the pop-up menu.

Note: This wasn't an issue before when the menu was on the top.

![image](https://github.com/cuzy-app/humhub-modules-clean-theme/assets/13114/8d370de1-9e91-47fb-a201-6f86669f49a9)

**Comment by WizWareTech** [comment]
Another issue:

When searching under the my spaces button, if there are a full screen of results:
1) They fall off the top edge and you can't see anything above
2) the input field is also off the top of the screen
3) the on screen keyboard disappears
4) when attempting to scroll the results up/down, the entire page moves instead of the pop-up result menu

Proposed solution:
limit results that show to allow the input field to remain on the screen at the top, and keep the on screen keyboard on screen so that characters can continue being typed in to narrow the search further. also, fix the scrolling so the results can be scrolled within the pop-up menu.

Note: This wasn't an issue before when the menu was on the top.

![image](https://github.com/cuzy-app/humhub-modules-clean-theme/assets/13114/a2b8517b-f89e-48ac-86c6-8f4a2540126d)

**Comment by marc-farre** [comment]
@WizWareTech thanks for reporting this issue.

Fixed in commits https://github.com/cuzy-app/humhub-modules-clean-theme/commit/58440abe6bffb64141a8dd4af4ea3e3353f665bb and https://github.com/cuzy-app/humhub-modules-clean-theme/commit/9d4d324f13f3aebad2a5599064d0110bec9b2a9c

Can you test the latest version by downloading https://github.com/cuzy-app/humhub-modules-clean-theme/archive/refs/heads/master.zip ? You might have to flush the cache after overwriting the new files on your server.

Let me know if it's OK now with it.
